### DISCUSSION TITLE
/hyndsight/graphics/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#51] [Announcements]
Twenty rules for good graphics 
 https://robjhyndman.com/hyndsight/graphics/

**Comment by robjhyndman** [comment]
```header
Originally posted by Steve P on 2010-08-06 14:12:54
````
Thanks for the tips. I was already doing most of these things, but it\'s
good to know that you agree.

**Reply by osorio** [reply]
Another Tufte rule, which may not conform to your 1.6 aspect ratio suggestion: Choose the aspect ratio so that the approximate average slope of the curve(s) is +/-1 (i.e. 45 degrees).  Limiting nearly vertical or horizontal slopes decreases possible confusion about what x and y values correspond to each point.

**Comment by robjhyndman** [comment]
```header
Originally posted by Petr on 2010-08-06 18:41:35
````
rule #1- vector graphics is good only if you do not work with extremely
large number of data points. A figure with millions of lines can be
easily as large as hundreds of Mb when ps or pdf format is used. Also
rendering of such image can take a long time Use of png IMO is a good
choice then.\
Petr

**Reply by robjhyndman** [reply]
```header
Originally posted on 2010-08-07 00:34:41
````
Hi Petr. Thanks for your comment. I agree \-- I also use png when I have
more than about 10000 points or lines.

**Comment by robjhyndman** [comment]
```header
Originally posted by Ben Bolker on 2010-08-07 17:51:06
````
\* Make as much text horizontal as possible (par(las=1), and possibly\
swapping x- and y-axes so that long labels sit along the y axis where\
they can be spelled out horizontally and not overlap)

\* eliminate redundant sets of axis labels for small multiple plots (à
la lattice and\
ggplot) if possible

\* the \'classic books\' aren\'t showing up for me. Tufte and Cleveland?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2010-08-07 22:42:30
````
The books will be invisible if you have an adblocker. Yes, Tufte and
Cleveland are the classics (imo).

**Comment by robjhyndman** [comment]
```header
Originally posted by Carlos on 2010-08-21 18:26:14
````
I don\'t understand \# 15. Should it be something like &quot;Plots should be
self-explanatory, so avoid excessively detailed captions&quot;?

There should also be rule \# 0: Have something to say. Too many charts
are pointless.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2010-08-22 07:29:39
````
My point in #15 is to avoid the situation where you have to search all
the surrounding text trying to figure out what the plot means, or what
the variables are. This should be contained in the plot itself. If the
plot is sufficiently self-explanatory that it doesn\'t need much of a
caption, that\'s great. But sometimes explanation is required, and then
it is better to put it in the caption rather than require the reader to
find the information somewhere else.

I agree with your rule #0!

**Comment by robjhyndman** [comment]
```header
Originally posted by Naomi B Robbins on 2011-12-09 23:47:56
````
Some of your advice contradicts that of Cleveland. For example,
Cleveland says,&quot;Avoid putting notes and keys inside the scale line
rectangle.&quot;  Creating More Effective Graphs includes an example where
the data and the key are difficult to distinguish since the key is
inside the plot area.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2011-12-10 00:14:17
````
If there is no room for a key inside the region without causing
confusion, then I wouldn\'t do it. But often (at least in my plots)
there are blank regions near one or more corners where a key can easily
fit without obscuring anything.  Then I think it is neater to put the
key inside. It is hard to have a consistent rule here. Perhaps we need a
meta-rule that says don\'t always follow the rules!

**Comment by robjhyndman** [comment]
```header
Originally posted by neilfws on 2013-08-16 00:51:04
````
R/ggplot2 uses a grey background by default but it is simple to remove.
Easiest way: + theme_bw()

**Comment by robjhyndman** [comment]
```header
Originally posted by Gaz on 2013-08-16 00:59:49
````
You instantly fail your students if they use a chart type you don\'t
happen to like? I bet they think you\'re an absolute jerk.

**Reply by robjhyndman** [reply]
```header
Originally posted on 2013-08-16 01:47:57
````
My students recognize the sarcasm, even if my readers do not.

**Comment by robjhyndman** [comment]
```header
Originally posted by molecule61 on 2014-04-17 18:37:00
````
Regarding rule #7: You need to have a better reason not to include zero
on your y axis than simply avoiding &quot;excessive&quot; whitespace. When
you\'re plotting data whose absolute value is meaningful, that
whitespace may be part of the value.

**Comment by robjhyndman** [comment]
```header
Originally posted by Jolicharts.com on 2014-09-01 08:47:54
````
Thanks for making quick tips. That will surely help people to graph a
lot better.

**Comment by robjhyndman** [comment]
```header
Originally posted by Jay Jang on 2014-09-23 22:52:28
````
Thanks for the tips. I\'ve been working in a financial securities firm
for about 10 months now, where I started my first full-time job as a
fresh graduate after the 2-year Korean military service that I had after
my graduation in Australia.

I am in a financial product planning team at a financial product
strategy department within the firm. I occasionally write up some
reports and they sometimes include few pie-charts and also the 3D
pie-charts that you said not to use in the rule #20.\
Due to lack of knowledge and experience in the field, may I ask you the
reasons not to use the pie-charts, sir?

**Reply by robjhyndman** [reply]
```header
Originally posted on 2014-09-23 23:57:23
````
Any book on statistical graphics should discuss this. Try authors such
as Bill Cleveland, Stephen Few, Naomi Robbins, \...

**Comment by robjhyndman** [comment]
```header
Originally posted by Joe Lotz on 2014-12-26 22:55:16
````
This post is an oldey but goody. I still reference people to read it!

**Comment by robjhyndman** [comment]
```header
Originally posted by Eva Wilson on 2019-03-07 14:58:56
````
I\'ll try to follow the rules.
